### PR TITLE
Docs additions:

### PR DESCRIPTION
**Pull request by holmanb**
```
Docs additions:
    
- Add service completion blockers to FAQ
- Update LXD config option names from deprecated names
- Add neovim configuration blog post
- Make docs test fail early on spelling error
- Update spelling filter
```
Example early exit success:
https://app.travis-ci.com/github/canonical/cloud-init/jobs/570863456

### CODE REVIEW

File: doc/rtd/topics/faq.rst [hunk 1]
```diff
@@ -151,6 +151,36 @@ provided to the system:
...
+- bugs in the kernel or drivers
+- bugs in external userspace tools that are called by cloud-init
+
+## Internal reasons:
+--------------------
+- a command in `bootcmd` or `runcmd` that never completes (ex: running
+  `cloud-init status --wait` will wait forever on itself and never complete)
+- nonstandard configurations that disable timeouts or set extremely high
+  values ("never" is used in a loose sense here)
+- cloud-init is not immune to bugs
```
Comment (TheRealFalcon) [inline doc/rtd/topics/faq.rst:182]: I don't think this line adds much. I think everybody knows any software can have bugs.

File: doc/rtd/topics/faq.rst [hunk 2]
```diff
@@ -151,6 +151,36 @@ provided to the system:
...
+```
+cloud-init status
+```
+
+To wait for clous-init to complete, run:
+```
+cloud-init status --wait
+```
+
+There are a number of reasons that cloud-init might "never complete. This list
```
Comment (TheRealFalcon) [inline doc/rtd/topics/faq.rst:167]: Did you mean to say "never" (in quotes) ? I don't think the quotes are necessary.

File: doc/rtd/topics/faq.rst [hunk 3]
```diff
@@ -212,17 +242,15 @@ launch this multiple times:
...
-| vendor data    | user.vendor-data    |
-+----------------+---------------------+
-| metadata       | user.meta-data      |
-+----------------+---------------------+
-| network config | user.network-config |
-+----------------+---------------------+
++----------------+---------------------------+
+| Data           | Config Option             |
++================+===========================+
+| user data      | cloud-init.user-data      |
```
Comment (TheRealFalcon) [inline doc/rtd/topics/faq.rst:248]: Now that I think of it, we might want to do a broader search of `user.user-data` and replace with `cloud-init.user-data`. I see one a few lines above this change and I know I had it in my tutorial too. Doesn't have to be this PR, but we should add a ticket for it if not.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Good changes. Comments inline.

File: doc/rtd/topics/faq.rst (hunk 3 above)
Comment (holmanb) [reply doc/rtd/topics/faq.rst:247]: Agreed. I just added a ticket to the backlog for this.

**Comment by TheRealFalcon** [comment]
This isn't quite rendering the way I'd expect either.
image

**Review by TheRealFalcon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**